lib/application/view-models/execution/collection_execution_vm.dart

```diff
@@ -3,70 +3,165 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
+  final CollectionServices collectionServices;
 
   late final List<Memo> _memos;
+  late final Collection _collection;
   final _executions = <MemoExecution>[];
   var _currentMemoStartDate = DateTime.now().toUtc();
 
   bool get _isFinished => _memos.length == _executions.length;
 
   @override
-  Future<void> markCurrentMemoDifficulty(MemoDifficulty difficulty) async {
+  void markCurrentMemoDifficulty(MemoDifficulty difficulty) {
     if (state is! LoadedCollectionExecutionState) {
       throw InconsistentStateError.viewModel('Cannot mark the current memo before finishing loading');
     }
 
+    state = (state as LoadedCollectionExecutionState).copyWith(markedAnswer: difficulty);
+  }
+
+  @override
+  Future<void> nextContents() async {
+    if (state is! LoadedCollectionExecutionState) {
+      throw InconsistentStateError.viewModel('Cannot request next memo contents before finishing loading');
+    }
+
     final loadedState = state as LoadedCollectionExecutionState;
 
+    // If there is a `markedAnswer`, we assume that this is a request to forward with the last marked difficulty
+    if (loadedState.markedAnswer != null) {
+      await _confirmMarkedMemo(loadedState.markedAnswer!);
+    } else {
+      // Otherwise, we expect it to be a request to display the question
+      final currentMemo = _memos[_executions.length];
+      state = loadedState.copyWith(isDisplayingQuestion: false, currentContents: currentMemo.rawAnswer);
+    }
+  }
+
+  /// Makes the according updates to the internal control properties of an execution
+  Future<void> _confirmMarkedMemo(MemoDifficulty answer) async {
+    final currentMemo = _memos[_executions.length];
+
     final execution = MemoExecution(
-      memoId: loadedState._currentMemo.id,
+      uniqueId: currentMemo.uniqueId,
       collectionId: collectionId,
       started: _currentMemoStartDate,
       finished: DateTime.now().toUtc(),
-      rawQuestion: loadedState.rawQuestion,
-      rawAnswer: loadedState.rawAnswer,
-      markedDifficulty: difficulty,
+      rawQuestion: currentMemo.rawQuestion,
+      rawAnswer: currentMemo.rawAnswer,
+      markedDifficulty: answer,
     );
     _executions.add(execution);
 
+    // If the `executions` update makes this `_isFinished` to true, we have to update this state to a
+    // `FinishedCollectionExecutionState`
     if (_isFinished) {
       state = LoadingCollectionExecutionState();
-      await _services.addExecutions(_executions, collectionId: collectionId);
-      state = FinishedCollectionExecutionState();
+      await executionServices.addExecutions(_executions, collectionId: collectionId);
+
+      final mappedExecutions = <MemoDifficulty, int>{
+        for (final difficulty in MemoDifficulty.values) difficulty: 0,
+      };
+
+      _executions.forEach((execution) {
+        final difficultyValue = mappedExecutions[execution.markedDifficulty];
+        mappedExecutions[execution.markedDifficulty] = difficultyValue! + 1;
```

Comment:
You probably tried it, but cant we just make `mappedExecutions[execution.markedDifficulty] += 1`?

Reply:
we sadly can't, but I thinlk there is a possibility of doing it by extending the operators, although I haven't tried